### PR TITLE
🐛 Fix jumping to top of page when next page loads

### LINKED ISSUE #32165: User is sent back to the top of the first article when next page loads
Issue:
When I approach the end of the first article and the start of the amp-next-page article, the user is sent back to the top of the first article.

To reproduce:

Search "CNET" in Google, click on an AMP page from SERP
Scroll to near end of article and into amp-next-page.
Notice that you are thrown back to top of the first article
Demo:

Via iPhone Safari browser: https://drive.google.com/file/d/1WrT03HMOhQ0Oaw2BMbjbtVWyiiKByRxT/view?usp=sharing
Via Chrome Emulator: https://drive.google.com/file/d/1xqu3BaJRXhGr3fT-zuFVT2VrdvZiXCsV/view?usp=sharing

(copies from https://github.com/ampproject/amphtml/issues/31951#issuecomment-760386354)

### PR DESCRIPTION
**Pull request by dmanek**
Fixes #32165

### CODE REVIEW

File: src/service/viewport/viewport-impl.js [hunk 1]
```diff
@@ -209,8 +209,14 @@ export class ViewportImpl {
     // https://github.com/ampproject/amphtml/issues/30838 for more details.
     // The solution is to make a "fake" scrolling API call.
     const isIframedIos = Services.platformFor(win).isIos() && isIframed(win);
-    if (isIframedIos) {
-      this.ampdoc.whenReady().then(() => win./*OK*/ scrollTo(-0.1, 0));
+    // Fix for https://github.com/ampproject/amphtml/issues/32165, we don't
+    // want to scroll if we're in a shadow doc.
+    const isShadowDoc =
+      this.ampdoc.getRootNode().nodeType !== /* DOCUMENT_NODE */ 9;
```
Comment (jridgewell) [inline src/service/viewport/viewport-impl.js:215]: Can we just check if `ampdoc.isSingleDoc()`?

Comment (dmanek) [reply src/service/viewport/viewport-impl.js]: Done.

Comment (jridgewell) [reply src/service/viewport/viewport-impl.js]: Actually, this was a question for @dvoytenko. 😬

Comment (dmanek) [reply src/service/viewport/viewport-impl.js]: Ah ok, will leave it as is unless @dvoytenko has a strong preference.

Comment (dvoytenko) [reply src/service/viewport/viewport-impl.js]: Oh, yeah. `isSingleDoc` works great for this. Good idea.

**Review by dvoytenko — APPROVED** [review]

**Review by jridgewell — APPROVED** [review]

**Pull request merged**